client/apis/volumegroupsnapshot/v1alpha1/types.go

```diff
@@ -0,0 +1,284 @@
+
+	// CreationTime is the timestamp when the point-in-time group snapshot is taken
+	// by the underlying storage system.
+	// If not specified, it may indicate that the creation time of the group snapshot
+	// is unknown.
+	// +optional
+	CreationTime *metav1.Time `json:"creationTime,omitempty" protobuf:"bytes,2,opt,name=creationTime"`
+
+	// ReadyToUse indicates if all the individual snapshots in the group are ready
+	// to be used to restore a volume.
+	// If not specified, it means the readiness of a group snapshot is unknown.
+	// +optional
+	ReadyToUse *bool `json:"readyToUse,omitempty" protobuf:"varint,3,opt,name=readyToUse"`
+
+	// Error is the last observed error during group snapshot creation, if any.
+	// This field could be helpful to upper level controllers (i.e., application
+	// controller) to decide whether they should continue on waiting for the group
+	// snapshot to be created based on the type of error reported.
+	// The snapshot controller will keep retrying when an error occurs during the
+	// group snapshot creation. Upon success, this error field will be cleared.
+	// +optional
+	Error *snapshotv1.VolumeSnapshotError `json:"error,omitempty" protobuf:"bytes,4,opt,name=error,casttype=VolumeSnapshotError"`
+
+	// VolumeSnapshotRefList is the list of volume snapshot references for this
+	// group snapshot.
+	// The maximum number of allowed snapshots in the group is 100.
+	// +optional
+	VolumeSnapshotRefList []core_v1.ObjectReference `json:"volumeSnapshotRefList,omitempty" protobuf:"bytes,5,opt,name=volumeSnapshotRefList"`
+}
+
+//+genclient
+//+k8s:deepcopy-gen=true
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//+kubebuilder:resource:scope=Namespaced,shortName=vgs
+//+kubebuilder:subresource:status
+
+// VolumeGroupSnapshot is a user's request for creating either a point-in-time
+// group snapshot or binding to a pre-existing group snapshot.
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+type VolumeGroupSnapshot struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   VolumeGroupSnapshotSpec   `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	Status VolumeGroupSnapshotStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// VolumeGroupSnapshotList contains a list of VolumeGroupSnapshot
+type VolumeGroupSnapshotList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VolumeGroupSnapshot `json:"items"`
+}
+
+//+genclient
+//+genclient:nonNamespaced
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeGroupSnapshotClass is the Schema for the volumegroupsnapshotclass API
```

Comment:
Change to this:
// VolumeGroupSnapshotClass specifies parameters that a underlying storage system uses when
// creating a volume group snapshot. A specific VolumeGroupSnapshotClass is used by specifying its
// name in a VolumeGroupSnapshot object.
// VolumeGroupSnapshotClasses are non-namespaced